init.cc

```diff
@@ -50,6 +50,10 @@ std::set<gms::inet_address> get_seeds_from_db_config(const db::config& cfg) {
         startlog.error("Use broadcast_address instead of listen_address for seeds list");
         throw std::runtime_error("Use broadcast_address for seeds list");
     }
+    if (!cfg.replace_node_first_boot().empty() && seeds.contains(broadcast_address)) {
+        startlog.error("Bad configuration: replace-node-first-boot is not allowed for seed nodes");
+        throw bad_configuration_error();
+    }
     if ((!cfg.replace_address_first_boot().empty() || !cfg.replace_address().empty()) && seeds.contains(broadcast_address)) {
```

Comment:
Why not include `!cfg.replace_node_first_boot().empty()` in this alternative (and modify the error contents)? It would be more consistent to either put all the options together since the existing options were already together.

Reply:
It's possible, but it will mean less churn once the support for the old options is dropped.
Also the inner logic will have to check `cfg.replace_node_first_boot().empty()` again to control the error message.